python/python/tests/test_async_client.py

```diff
@@ -10254,3 +10255,155 @@ async def test_script_large_keys_and_args(self, request, cluster_mode, protocol)
             == key.encode()
         )
         await glide_client.close()
+
+    @pytest.mark.parametrize("cluster_mode", [True, False])
+    @pytest.mark.parametrize("protocol", [ProtocolVersion.RESP2, ProtocolVersion.RESP3])
+    async def test_script_exists(self, glide_client: TGlideClient):
+        script1 = Script("return 'Hello'")
+        script2 = Script("return 'World'")
+
+        # Load the scripts
+        await glide_client.invoke_script(script1)
+        await glide_client.invoke_script(script2)
+
+        # Get the SHA1 digests of the scripts
+        sha1_1 = script1.get_hash()
+        sha1_2 = script2.get_hash()
+        non_existent_sha1 = "0" * 40  # A SHA1 that doesn't exist
+
+        # Check existence of scripts
+        result = await glide_client.script_exists([sha1_1, sha1_2, non_existent_sha1])
+
+        assert result == [True, True, False]
+
+    @pytest.mark.parametrize("cluster_mode", [True, False])
+    @pytest.mark.parametrize("protocol", [ProtocolVersion.RESP2, ProtocolVersion.RESP3])
+    async def test_script_flush(self, glide_client: TGlideClient):
+        # Load a script
+        script = Script("return 'Hello'")
+        await glide_client.invoke_script(script)
+
+        # Check that the script exists
+        assert await glide_client.script_exists([script.get_hash()]) == [True]
+
+        # Flush the script cache
+        assert await glide_client.script_flush() == OK
+
+        # Check that the script no longer exists
+        assert await glide_client.script_exists([script.get_hash()]) == [False]
+
+        # Test with ASYNC mode
+        await glide_client.invoke_script(script)
+        assert await glide_client.script_flush(FlushMode.ASYNC) == OK
+        assert await glide_client.script_exists([script.get_hash()]) == [False]
+
+    @pytest.mark.parametrize("cluster_mode", [True, False])
+    @pytest.mark.parametrize("protocol", [ProtocolVersion.RESP2, ProtocolVersion.RESP3])
+    async def test_script_kill(
+        self, request, cluster_mode, protocol, glide_client: TGlideClient
```

Comment:
I think you need to test cluster client with single node route and with multi node route both